core-common/src/main/java/org/mapstruct/Context.java

```diff
@@ -78,6 +80,61 @@
  * }
  * </code>
  * </pre>
+ * <p>
+ * <strong>Example 2:</strong> Using an {@code @Context} parameter with a type that provides its own {@code @}
+ * {@link BeforeMapping} methods to handle cycles in Graph structures:
+ *
+ * <pre>
+ * <code>
+ * // type of the context parameter
+ * public class CyclicGraphContext {
+ *     private Map&lt;Object, Object&gt; knownInstances = new IdentityHashMap&lt;&gt;();
+ *
+ *     &#64;BeforeMapping
+ *     public &lt;T extends NodeDto&gt; T getMappedInstance(Object source, &#64;TargetType Class&lt;T&gt; targetType) {
+ *         return (T) knownInstances.get( source );
+ *     }
+ *
+ *     &#64;BeforeMapping
+ *     public void storeMappedInstance(Object source, &#64;MappingTarget NodeDto target) {
+ *         knownInstances.put( source, target );
+ *     }
+ * }
+ *
+ * &#64;Mapper
+ * public interface GraphMapper {
+ *     NodeDto toNodeDto(Node node, &#64;Context CyclicGraphContext cycleContext);
+ * }
+ *
+ *
+ * // generates:
+ *
+ * public NodeDto toNodeDto(Node node, CyclicGraphContext cycleContext) {
+ *     NodeDto target = cycleContext.getMappedInstance( node, NodeDto.class );
+ *     if ( target != null ) {
+ *         return target;
+ *     }
+ *
+ *     if ( node == null ) {
+ *         return null;
+ *     }
+ *
+ *     NodeDto nodeDto = new NodeDto();
+ *
+ *     cycleContext.storeMappedInstance( node, nodeDto );
```

Comment:
How is it ensured that `getMappedInstance()` is called on line 113 and here then `storeMappedInstance()`?